c10/core/SymbolicIntNode.h

```diff
@@ -13,7 +13,21 @@ class C10_API SymbolicIntNode
  public:
   c10::SymInt toSymInt();
   virtual ~SymbolicIntNode(){};
+  // these could be pure virtual when we implement LTC versions
+  virtual SymbolicIntNode* add(SymbolicIntNode* other) { TORCH_CHECK(false, "NYI"); };
+  virtual SymbolicIntNode* sub(SymbolicIntNode* other) { TORCH_CHECK(false, "NYI"); };
+  virtual SymbolicIntNode* mul(SymbolicIntNode* other) { TORCH_CHECK(false, "NYI"); };
+  virtual SymbolicIntNode* div(SymbolicIntNode* other) { TORCH_CHECK(false, "NYI"); };
+  virtual SymbolicIntNode* mod(SymbolicIntNode* other) { TORCH_CHECK(false, "NYI"); };
+  virtual SymbolicIntNode* eq(SymbolicIntNode* other) { TORCH_CHECK(false, "NYI"); };
+  virtual SymbolicIntNode* gt(SymbolicIntNode* other) { TORCH_CHECK(false, "NYI"); };
+  virtual SymbolicIntNode* lt(SymbolicIntNode* other) { TORCH_CHECK(false, "NYI"); };
+  virtual SymbolicIntNode* wrap(int64_t num) { TORCH_CHECK(false, "NYI"); };
+  virtual bool bool_() { TORCH_CHECK(false, "NYI"); };
+  virtual int64_t int_() { TORCH_CHECK(false, "NYI"); }
+  virtual std::string str() { TORCH_CHECK(false, "NYI"); };
   virtual std::ostream& operator<<(std::ostream& os) {
```

Comment:
does this still need to be virtual

Reply:
I think it does the LTC impl won't have a python object to redispatch or fallback to

Reply:
but they'll implement str instead?

Reply:
OH, just realized what you meant ... operator<< does *NOT* to be virtual me thinks :)